### PR TITLE
Adding Buffering Docs to v13

### PR DESCRIPTION
**Pull request by FancyFane**
Signed-off-by: FancyFane <fane@planetscale.com>

Adding documentation for Buffering to the v13 Vitess documentation; after closing out the previous pull request which was against the non-versioned edition of the website repo. 

Also adding an additional entry in the .gitignore file for the hugo lock file which gets generated locally.

### CODE REVIEW

**Comment by deepthi** [comment]
@FancyFane can you resolve the conflicts while we wait for a review?

File: content/en/docs/13.0/reference/features/vtgate-buffering.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+First, let us cover the parameters that need to be set in VTGate to enable
+buffering:
+
+  * `-enable_buffer`:  Enables buffering.  **Not enabled by default**
+  * `-enable_buffer_dry_run`:  Enable logging of if/when buffering would
+  trigger, without actually buffering anything. Useful for testing.
+  Default: `false`
+  * `-buffer_implementation`:  Default: `healthcheck`.  More consistent results
+  have been seen with `keyspace_events` and the default will change to
+  `keyspace_events` in future releases.
```
Comment (mattlord) [inline content/en/docs/13.0/reference/features/vtgate-buffering.md]: These are the 13.0 docs and the default has been changed, no?

Comment (FancyFane) [reply content/en/docs/13.0/reference/features/vtgate-buffering.md]: Correct, this also reminded me to adjust the vtgate program documentation in terms of flags available and their default settings.

**Review by vmg — APPROVED** [review]
This is very neat. I really like the buffering scenario examples!

File: content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+```
+Terminal 1
+    $ git clone git@github.com:planetscale/vitess.git
+    $ cd vitess
+    $ make build
+    $ export PATH=${PATH}:$(pwd)/bin
+    $ cd examples/local
+    $ source env.sh
+    $ ./101_initial_cluster.sh
+```
```
Comment (deepthi) [inline content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md:48]: Can we make this a bit more general?
We have the `Getting Started` guides for people to install or build from source.
Depending on their method, they shouldn't even need to clone the source code repo.
For example, this guide https://github.com/vitessio/website/blob/prod/content/en/docs/13.0/get-started/local.md?plain=1#L85-L89

File: content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+
+#### Terminal 2:
+
+4.) In a NEW terminal window download and configure gateslap. This utility will
+be used to simulate traffic on Vitess. The virtualenv and source commands are
+optional:
+
+```
+Terminal 2
+    $ git clone https://github.com/FancyFane/gateslap.git
```
Comment (deepthi) [inline content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md:82]: FancyFane -> planetscale

**Review by deepthi — COMMENTED** [review]
I was reading this guide on the website, and it's pretty great.
A couple of suggestions for improvement.

File: content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md (hunk 2 above)
Comment (FancyFane) [reply content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md:48]: I agree, we can re-word this section to mention we are starting from the 101_inital_cluster.sh build, and we can provide them a link to the getting started section if it is needed.

File: content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md (hunk 3 above)
Comment (FancyFane) [reply content/en/docs/13.0/user-guides/configuration-advanced/buffering-scenarios.md:82]: Yes, this was leftover from when the repo was within my personal GitHub. I will work to get this reference fixed as well.

**Pull request merged**